WcaOnRails/app/jobs/submit_report_nag_job.rb

```diff
@@ -3,20 +3,45 @@
 class SubmitReportNagJob < SingletonApplicationJob
   queue_as :default
 
-  def nag_needed(competition)
+  def nag_needed?(competition)
     (competition.delegate_report.nag_sent_at || competition.end_date) <= 8.days.ago
   end
 
+  def reminder_needed?(competition)
+    return unless competition.end_date
+    return if competition.delegate_report.reminder_sent_at
+
+    # Send a reminder 6 days after the end of the competition
+    6.days.since(competition.end_date).today?
```

Comment:
Is there any way that the logic behind this check could be "unified" with the `nag_needed?` above?

In particular, the nag check is very "dense", while this reminder check is spread over three (non-comment) lines. I definitely prefer the thorough line-by-line check but in general I think it would be great for code maintainability to have these two methods look (almost) the same.

Reply:
Fixed :wrench: